mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,487 @@
+            sk_metrics.ConfusionMatrixDisplay(confusion_matrix=confusion_matrix).plot()
+
+        if hasattr(sk_metrics, 'ConfusionMatrixDisplay'):
+            self._log_image_artifact(
+                plot_confusion_matrix, "confusion_matrix",
+            )
+
+        self._log_metrics()
+        self._log_model_explainability()
+        return EvaluationResult(self.metrics, self.artifacts)
+
+    def _evaluate_regressor(self):
+        self.y_pred = self.model.predict(self.X)
+        self.metrics["example_count"] = len(self.X)
+        self.metrics["mean_absolute_error"] = sk_metrics.mean_absolute_error(self.y, self.y_pred)
+        self.metrics["mean_squared_error"] = sk_metrics.mean_squared_error(self.y, self.y_pred)
+        self.metrics["root_mean_squared_error"] = math.sqrt(self.metrics["mean_squared_error"])
+        self.metrics['sum_on_label'] = sum(self.y)
+        self.metrics['mean_on_label'] = self.metrics['sum_on_label'] / self.metrics["example_count"]
+        self.metrics['r2_score'] = sk_metrics.r2_score(self.y, self.y_pred)
+        self.metrics['max_error'] = sk_metrics.max_error(self.y, self.y_pred)
+        self.metrics['mean_absolute_percentage_error'] = \
+            sk_metrics.mean_absolute_percentage_error(self.y, self.y_pred)
+
+        self._log_metrics()
+        self._log_model_explainability()
+        return EvaluationResult(self.metrics, self.artifacts)
+
+    def evaluate(
+        self,
+        model: "mlflow.pyfunc.PyFuncModel",
+        model_type,
+        dataset,
+        run_id,
+        evaluator_config,
+        **kwargs,
+    ):
+        with TempDir() as temp_dir:
+            self.client = mlflow.tracking.MlflowClient()
+
+            self.temp_dir = temp_dir
+            self.model = model
+            self.model_type = model_type
+            self.dataset = dataset
+            self.run_id = run_id
+            self.evaluator_config = evaluator_config
+            self.dataset_name = dataset.name
+            self.feature_names = dataset.feature_names
+
+            model_loader_module, raw_model, predict_fn, predict_proba_fn = \
+                _extract_raw_model_and_predict_fn(model)
+            self.model_loader_module = model_loader_module
+            self.raw_model = raw_model
+            self.predict_fn = predict_fn
+            self.predict_proba_fn = predict_proba_fn
+
+            X, y = dataset._extract_features_and_labels()
+            self.X = X
+            self.y = y
+            self.metrics = EvaluationMetrics()
+            self.artifacts = {}
+
+            infered_model_type = _infer_model_type_by_labels(y)
+
+            if model_type != infered_model_type:
+                _logger.warning(
+                    f"According to the evaluation dataset label values, the model type looks like "
+                    f"{infered_model_type}, but you specified model type {model_type}. Please "
+                    f"check you set model type or evaluation dataset correctly."
```

Comment:
Awesome warning message, @WeichenXu123 !

```suggestion
                    f"{infered_model_type}, but you specified model type {model_type}. Please "
                    f"verify that you set the `model_type` and `dataset` arguments correctly."
```